kf/kf_unix.go

```diff
@@ -232,9 +235,48 @@ func (b *BPF) LoadTCAttachProgram(ifaceName, direction string) error {
 		if iface.Name == ifaceName && qdisc.Kind == "clsact" {
 			clsactFound = true
 		}
+		if iface.Name == ifaceName && qdisc.Kind == "htb" {
+			htbFound = true
+			htbHandle = qdisc.Msg.Handle
+		}
+		if iface.Name == ifaceName && qdisc.Kind == "ingress" {
+			ingressFound = true
+			ingressHandle = qdisc.Msg.Handle
+		}
 	}
 
-	if !clsactFound {
+	bpfRootProg := b.ProgMapCollection.Programs[b.Program.EntryFunctionName]
+
+	var parent uint32
+	var filter tc.Object
+
+	if clsactFound {
+		if direction == models.IngressType {
+			parent = tc.HandleMinIngress
+		} else if direction == models.EgressType {
+			parent = tc.HandleMinEgress
+		}
+		progFD := uint32(bpfRootProg.FD())
+		// Netlink attribute used in the Linux kernel
+		bpfFlag := uint32(tc.BpfActDirect)
+
+		filter = tc.Object{
+			Msg: tc.Msg{
+				Family:  unix.AF_UNSPEC,
+				Ifindex: uint32(iface.Index),
+				Handle:  0,
+				Parent:  core.BuildHandle(tc.HandleRoot, parent),
+				Info:    0x300,
+			},
+			Attribute: tc.Attribute{
+				Kind: "bpf",
+				BPF: &tc.Bpf{
+					FD:    &progFD,
+					Flags: &bpfFlag,
+				},
+			},
+		}
+	} else if !clsactFound && !ingressFound && !htbFound {
```

Comment:
IMHO For clarity, this condition block can be relocated to the end as part of the else statement.